rust-runtime/aws-smithy-http/src/operation/error.rs

```diff
@@ -185,3 +186,21 @@ impl Error for BuildError {
         }
     }
 }
+
+impl From<BuildError> for DeserializeError {
+    fn from(build_error: BuildError) -> Self {
+        DeserializeError::custom_source(
+            "deserialization failed because a required field was missing",
+            build_error,
+        )
+    }
+}
+
+#[cfg(feature = "event-stream")]
+impl From<BuildError> for aws_smithy_eventstream::error::Error {
+    fn from(build_error: BuildError) -> Self {
+        aws_smithy_eventstream::error::Error::unmarshalling(format!(
+            "error unmarshalling failed because a required field was missing: {build_error}"
```

Comment:
How does this error know the underlying cause was a required field?

Reply:
Currently, that's the only reason that this occurs. I changed this from a more generic message b/c of [Russell's comment here](https://github.com/awslabs/smithy-rs/pull/2916#discussion_r1294734978). I'm open to suggestions. Another option is that I code-generate a `map_err` function for each place we need to convert this, but I wanted to avoid adding so many new lines to the SDKs.